### ISSUE TITLE
How to install flow typings?

### ISSUE THREAD

**Issue by monovertex** [pichillilorenzo/JavaScriptEnhancements#48]
I'm trying to follow the instructions at https://github.com/pichillilorenzo/JavaScriptEnhancements/wiki/Adding-new-definitions-with-flow-typed, but I'm a bit stumped.

I do not have the `Flow` menu entry and using `flow-typed install` manually in my project won't work, as I don't have flow as a dependency to the project itself.

Is there any other package supposed to be installed to get the mentioned menu entry?

**Comment by pichillilorenzo** [comment]
did you create the project using the plugin? This menu option is available only with javascript enhancements projects.. if you want use it in an existing project, follow these simple steps  https://github.com/pichillilorenzo/JavaScriptEnhancements/wiki/Using-it-with-an-existing-project

**Comment by monovertex** [comment]
Yes, I did follow the instructions and installed it for an existing project:

![image](https://user-images.githubusercontent.com/1837605/38993183-978843ce-43eb-11e8-8aa9-b18cb52069e3.png)

Here are the menu entries I have:

![image](https://user-images.githubusercontent.com/1837605/38993299-e2300f56-43eb-11e8-8de6-a4d6eed235c4.png)

**Comment by pichillilorenzo** [comment]
Are there errors in the Sublime Text console?? what kind of project did you create? an empty project?

**Comment by pichillilorenzo** [comment]
this is weird.. there is no error about that.. if you go to the plugin folder, can you find the `Main_disabled.sublime-menu` file like here https://github.com/pichillilorenzo/JavaScriptEnhancements/tree/master/src/listeners/flow ??

The plugin, when it detects a javascript enhancements project, enables this file renaming it into `Main.sublime-menu` in order to get seen in the sublime menu. Also, I re-tested it on linux and it works.

If you create another project, can you see it or not??

**Comment by monovertex** [comment]
Indeed, I can see it for a new, empty project. Any idea what could block the enabling of this menu?

**Comment by pichillilorenzo** [comment]
open your project and then the sublime text console. 
Send these commands (line by line!) to the console and tell me what is the output:
```python
JavaScriptEnhancements = __import__('JavaScript Enhancements')
util = JavaScriptEnhancements.src.libs.util
util.is_javascript_project()
JavascriptEnhancementsEnableFlowMenuEventListener = JavaScriptEnhancements.src.listeners.flow.JavascriptEnhancementsEnableFlowMenuEventListener
JavascriptEnhancementsEnableFlowMenuEventListener.path
JavascriptEnhancementsEnableFlowMenuEventListener.path_disabled
import os
os.path.isfile(JavascriptEnhancementsEnableFlowMenuEventListener.path)
os.path.isfile(JavascriptEnhancementsEnableFlowMenuEventListener.path_disabled)
```

Thanks

**Comment by monovertex** [comment]
```
>>> JavaScriptEnhancements = __import__('JavaScript Enhancements')
>>> util = JavaScriptEnhancements.src.libs.util
>>> util.is_javascript_project()
False
>>> JavascriptEnhancementsEnableFlowMenuEventListener = JavaScriptEnhancements.src.listeners.flow.JavascriptEnhancementsEnableFlowMenuEventListener
>>> JavascriptEnhancementsEnableFlowMenuEventListener.path
'/home/cosmin/.config/sublime-text-3/Packages/JavaScript Enhancements/src/listeners/flow/Main.sublime-menu'
>>> JavascriptEnhancementsEnableFlowMenuEventListener.path_disabled
'/home/cosmin/.config/sublime-text-3/Packages/JavaScript Enhancements/src/listeners/flow/Main_disabled.sublime-menu'
>>> import os
>>> os.path.isfile(JavascriptEnhancementsEnableFlowMenuEventListener.path)
False
>>> os.path.isfile(JavascriptEnhancementsEnableFlowMenuEventListener.path_disabled)
True
```
Apparently it doesn't see the project as being a JS one. I'm not sure why is that. Is there anything else required for the project to be considered like that except the .je-project-settings folder?

**Comment by monovertex** [comment]
Ah, I figured it out. I looked at the implementation of the `is_javascript_project` method and found out that you're expecting the project file to be **in** the project.

I have mine outside of it, in another folder. However, it doesn't go into the else branch, to check for `window.folders()`, as it would find the javascript project folder then.

**Comment by monovertex** [comment]
Yes, but as I said, Sublime doesn't have a requirement to have the .sublime-project file in the main folder. It can be anywhere, so I don't think it's a good idea that the plugin just assumes that. It should check the window.folders anyway, if didn't find the project file.

**Comment by pichillilorenzo** [comment]
So I can try to look at the root of each window.folders() for a `.je-project-settings` folder, but it will stop on the first one that it will find

**Comment by monovertex** [comment]
@pichillilorenzo , I opened PR #49, with my proposal for the update of this method. With it the plugin behaves as expected and adds the `Flow` submenu in `Tools`.

**Comment by monovertex** [comment]
However, now the `Add Flow definition` action doesn't work. The input appears as it should, I search for the package I want to add typings for, it says `Searching for '...' definitions` and then it disappears and nothing happens. I don't get a list of typings and I don't get the "not found" message. I tried to search typings for `ember` and `express`, same outcome for both of them.

**Comment by monovertex** [comment]
Thanks!
